x-pack/legacy/plugins/index_management/static/ui/components/mappings_editor.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { EuiCodeEditor } from '@elastic/eui';
+
+interface Props {
+  setGetDataHandler: (handler: () => { isValid: boolean; data: Mappings }) => void;
+  FormattedMessage: typeof ReactIntl.FormattedMessage;
+  defaultValue?: Mappings;
+}
+
+export interface Mappings {
+  [key: string]: any;
+}
+
+export const MappingsEditor = ({
+  setGetDataHandler,
+  FormattedMessage,
+  defaultValue = {},
+}: Props) => {
+  const [mappings, setMappings] = useState<string>(JSON.stringify(defaultValue, null, 2));
+
+  const getFormData = () => {
+    try {
+      const parsed: Mappings = JSON.parse(mappings);
+      return {
+        data: parsed,
+        isValid: true,
+      };
+    } catch (e) {
+      return {
+        isValid: false,
+        data: {},
+      };
+    }
+  };
+
+  useEffect(() => {
+    setGetDataHandler(getFormData);
+  }, [mappings]);
+
+  return (
+    <EuiCodeEditor
+      mode="json"
+      theme="textmate"
+      width="100%"
+      value={mappings}
+      setOptions={{
+        showLineNumbers: false,
+        tabSize: 2,
+        maxLines: Infinity,
+      }}
+      editorProps={{
+        $blockScrolling: Infinity,
+      }}
+      showGutter={false}
+      minLines={6}
+      aria-label={
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor"
```

Comment:
nit: should the `id` be a little more specific? 

https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#naming-convention

Reply:
What do you suggest @alisonelizabeth ? As this is a reusable component across features and apps (the Machine learning team might use it), I did not want to be too specific.

Reply:
@sebelga I was more concerned that the id doesn't end with a type - maybe something like `xpack.idxMgmt.mappingsEditor.mappingsEditorAriaLabel`?

Reply:
Ah ok, I will update the i18n key